File: comscore/src/api/ComscoreConfiguration.ts
export enum ComscoreUserConsent {
  denied = '0',
  granted = '1',
  unknown = '-1',
}

export enum ComscoreUsagePropertiesAutoUpdateMode {
  foregroundOnly = 'foregroundOnly',
  foregroundAndBackground = 'foregroundAndBackground',
  disabled = 'disabled',
}

export interface ComscoreConfiguration {
  /**
   * Also known as the c2 value
   */
  publisherId: string;
  applicationName: string;
  userConsent: ComscoreUserConsent;
  /**
   * Defaults to foregroundOnly if none is specified. If your app has some background experience, use foregroundAndBackground.
   */
  usagePropertiesAutoUpdateMode?: ComscoreUsagePropertiesAutoUpdateMode;
  debug?: boolean;
}

File: comscore/android/src/main/java/com/theoplayercomscore/integration/ComscoreConfiguration.kt
package com.theoplayercomscore.integration

data class ComscoreConfiguration(
  val publisherId: String,
  val applicationName: String,
  val usagePropertiesAutoUpdateMode: Int,
  val userConsent: String,
  val secureTransmission: Boolean,
  val childDirected: Boolean,
  val debug: Boolean
)
